server/pulp/plugins/util/publish_step.py

```diff
@@ -562,6 +563,77 @@ def process_lifecycle(self):
         return self._build_final_report()
 
 
+class UnitModelPluginStep(PluginStep):
+    """
+    Useful for performing actions on units that are defined as mongoengine models.
+    """
+    def __init__(self, step_type, model_classes, repo_content_unit_q=None, repo=None, conduit=None,
+                 config=None, working_dir=None, plugin_type=None, **kwargs):
+        """
+        :param step_type: The id of the step this processes
+        :type  step_type: str
+        :param model_classes:   list of ContentUnit subclasses that should be queried
+        :type  model_classes:   list
+        :param repo_content_unit_q: optional Q object that will be applied to the queries performed
+                                    against each ContentUnit class that gets passed in
+        :type  repo_content_unit_q: mongoengine.Q
+        :param repo: The repo being worked on
+        :type  repo: pulp.plugins.model.Repository
+        :param conduit: The conduit for the repo
+        :type  conduit: a conduit from pulp.plugins.conduits
+        :param config: The configuration
+        :type  config: PluginCallConfiguration
+        :param working_dir: The temp directory this step should use for processing
+        :type  working_dir: str
+        :param plugin_type: The type of the plugin
+        :type  plugin_type: str
+        """
+        super(UnitModelPluginStep, self).__init__(step_type, repo, conduit, config, working_dir,
```

Comment:
Not a change/concer of this PR, but what can be done to remove conduit from the call signature of `PluginStep`. The all-mongoengine objects shouldn't use the conduit. What do you think?